### PR TITLE
DbService CreateUserTests

### PR DESCRIPTION
**Pull request by LK777LK**

### CODE REVIEW

**Review by DanilFadeeev — APPROVED** [review]

File: Services/DataBaseService/Database/Models/DbUser.cs [hunk 1]
```diff
@@ -1,16 +1,70 @@
...
 using Microsoft.EntityFrameworkCore.Metadata.Builders;
 using System;
+using System.Diagnostics.CodeAnalysis;
 
 namespace DataBaseService.Database.Models
 {
-    public class DbUser
+    public class DbUser : IEquatable<DbUser>
     {
+        public static bool operator ==(DbUser first, DbUser second)
```
Comment (nmamro) [inline Services/DataBaseService/Database/Models/DbUser.cs]: Why you updated the class (and not only this one), will anyone need comparison anywhere apart from tests?

File: Common/DTO/User.cs [hunk 2]
```diff
@@ -2,12 +2,51 @@
 
 namespace DTO
 {
-    public class User
+    public class User : IEquatable<User>
```
Comment (nmamro) [inline Common/DTO/User.cs]: Don't need this interface

File: Common/DTO/User.cs [hunk 3]
```diff
@@ -2,12 +2,51 @@
...
     {
         public Guid Id { get; set; }
         public string Email { get; set; }
         public string FirstName { get; set; }
         public string LastName { get; set; }
         public DateTime Birthday { get; set; }
+
+        public bool Equals(User other)
```
Comment (nmamro) [inline Common/DTO/User.cs]: I think it's better to make checker in tests class, remove this code from data transfer objects please

**Review by nmamro — CHANGES_REQUESTED** [review]

File: Common/DTO/User.cs (hunk 3 above)
Comment (nmamro) [reply Common/DTO/User.cs]: Then you will have `AreUsersEqual()` in tests and you will call `Assert.IsTrue()` for example

File: Tests/DatabaseServiceTests/Mappers/UserMapperTests.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+        IUserMapper mapper;
+
+        User user;
+        DbUser dbUser;
+
+        UserCredential credential;
+        DbUserCredential dbCredential;
+
+        [SetUp]
```
Comment (ifelseelif) [inline Tests/DatabaseServiceTests/Mappers/UserMapperTests.cs]: All these objects will be initialized every time. Do you need it ?

File: Services/DataBaseService/Database/Models/DbUser.cs [hunk 5]
```diff
@@ -1,16 +1,70 @@
 ﻿using Microsoft.EntityFrameworkCore;
 using Microsoft.EntityFrameworkCore.Metadata.Builders;
 using System;
+using System.Diagnostics.CodeAnalysis;
 
 namespace DataBaseService.Database.Models
 {
-    public class DbUser
+    public class DbUser : IEquatable<DbUser>
```
Comment (ifelseelif) [inline Services/DataBaseService/Database/Models/DbUser.cs]: I do not understand why you need it ?

**Review by VLiamin — APPROVED** [review]

**Review by ifelseelif — APPROVED** [review]

**Pull request closed without merging**